checkstyle/src/main.rs

```diff
@@ -0,0 +1,212 @@
+use clap::Parser;
+use regex::Regex;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::process;
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[arg(short, long)]
+    target_path: String,
+
+    #[arg(short, long)]
+    regex_file_path: String,
+
+    #[arg(short, long)]
+    include: Option<Vec<String>>,
+    #[arg(short, long)]
+    exclude: Option<Vec<String>>,
+}
+
+fn main() -> Result<(), std::io::Error> {
```

Comment:
Now would also be a good time for us to create another file that contains all of the functions relevant to the include and excluding logic verification. Our main function will call those functions.

We can also move the relevant unit tests to the files that contain those functions.